tests/api.c

```diff
@@ -3314,6 +3318,294 @@ static int test_wc_Sha384Final (void)
+    Des3 des;
+
+    #ifndef WOLFSSL_SMALL_STACK
+        byte cipher[24];
+        byte plain[24];
+    #elif defined(WOLFSSL_SMALL_STACK)
+        byte* cipher = (byte*) malloc(sizeof(byte) * 24);
+        byte* plain = (byte*) malloc(sizeof(byte) * 24);
+    #endif
+
+    int ret;
+
+    const byte key[] =
+    {
+        0x01,0x23,0x45,0x67,0x89,0xab,0xcd,0xef,
+        0xfe,0xde,0xba,0x98,0x76,0x54,0x32,0x10,
+        0x89,0xab,0xcd,0xef,0x01,0x23,0x45,0x67
+    };
+
+    const byte iv[] =
+    {
+        0x12,0x34,0x56,0x78,0x90,0xab,0xcd,0xef,
+        0x01,0x01,0x01,0x01,0x01,0x01,0x01,0x01,
+        0x11,0x21,0x31,0x41,0x51,0x61,0x71,0x81
+    };
+
+    const byte vector[] = { /* "Now is the time for all " w/o trailing 0 */
+        0x4e,0x6f,0x77,0x20,0x69,0x73,0x20,0x74,
+        0x68,0x65,0x20,0x74,0x69,0x6d,0x65,0x20,
+        0x66,0x6f,0x72,0x20,0x61,0x6c,0x6c,0x20
+    };
+
+    printf(testingFmt, "wc_Des3_CbcEncrypt()");
+
+    AssertIntEQ(wc_Des3_SetKey(&des, key, iv, DES_ENCRYPTION), 0);
+
+    ret = wc_Des3_CbcEncrypt(&des, cipher, vector, 24);
+
+    AssertIntEQ(wc_Des3_SetKey(&des, key, iv, DES_DECRYPTION), 0);
+
+    if (ret == 0) {
+        ret = wc_Des3_CbcDecrypt(&des, plain, cipher, 24);
+    }
+
+    AssertIntEQ(XMEMCMP(plain, vector, 24), 0);
+
+    /* Pass in bad args. */
+    if (ret == 0) {
+        ret = wc_Des3_CbcEncrypt(NULL, cipher, vector, 24);
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcEncrypt(&des, NULL, vector, 24);
+        }
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcEncrypt(&des, cipher, NULL, sizeof(vector));
+        }
+        if (ret != BAD_FUNC_ARG) {
+            printf("\nReturn code: %d\n", ret);
+            ret = SSL_FAILURE;
+        } else {
+        ret = 0;
+        }
+    }
+
+    if (ret == 0) {
+        ret = wc_Des3_CbcDecrypt(NULL, plain, cipher, 24);
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcDecrypt(&des, NULL, cipher, 24);
+        }
+        if (ret == BAD_FUNC_ARG) {
+            ret = wc_Des3_CbcDecrypt(&des, plain, NULL, 24);
+        }
+        if (ret != BAD_FUNC_ARG) {
+            printf("\nReturn code: %d\n", ret);
+            ret = SSL_FAILURE;
+        } else {
+            ret = 0;
+        }
+    }
+
+    printf(resultFmt, ret == 0 ? passed : failed);
+
+    #ifdef WOLFSSL_SMALL_STACK
+        XFREE(plain, NULL, DYNAMIC_TYPE_NONE);
+        XFREE(cipher, NULL, DYNAMIC_TYPE_CIPHER);
+    #endif
+#endif
+    return 0;
+
+} /* END wc_Des3_CbcEncrypt */
+
+/*
+ *  Unit test for wc_Des3_CbcEncryptWithKey and wc_Des3_CbcDecryptWithKey
+ */
+static int test_wc_Des3_CbcEncryptDecryptWithKey (void)
+{
+#ifndef NO_DES3
+
+    word32 vectorSz, cipherSz;
+    int ret;
+
+    #ifndef WOLFSSL_SMALL_STACK
+        byte cipher[24];
+        byte plain[24];
+    #elif defined(WOLFSSL_SMALL_STACK)
+        byte* cipher = (byte*) malloc(sizeof(byte) * 24);
+        byte* plain = (byte*) malloc(sizeof(byte) * 24);
```

Comment:
Like above, these should probably be XMALLOC() calls instead of malloc(), for portability.